src/detect-isdataat.c

```diff
@@ -584,4 +681,41 @@ void DetectIsdataatRegisterTests(void)
     UtRegisterTest("DetectIsdataatTestPacket02", DetectIsdataatTestPacket02);
     UtRegisterTest("DetectIsdataatTestPacket03", DetectIsdataatTestPacket03);
 }
+
+static int DetectAbsentTestParse01(void)
+{
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    FAIL_IF(de_ctx == NULL);
+    de_ctx->flags |= DE_QUIET;
+
+    Signature *s = DetectEngineAppendSig(de_ctx,
+            "alert http any any -> any any "
+            "(msg:\"invalid absent only with negated content\"; http.user_agent; "
+            "absent; content:!\"one\"; sid:2;)");
+    FAIL_IF(s != NULL);
```

Comment:
Why is it designed to fail with any content following `absent`? If the `http.user_agent` is absent, nothing else can be matched against unless it's a sticky buffer?

Reply:
Yes `absent;` means only match if the buffer is absent, so cannot have additional tests on it.

Otherwise, we can use `absent: or_else;`